pkg/debug/portforward.go

```diff
@@ -0,0 +1,44 @@
+package debug
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/openshift/odo/pkg/occlient"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	k8sgenclioptions "k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+)
+
+type PortForwarder interface {
+	ForwardPorts(method string, url *url.URL, ports []string, stopChan, readyChan chan struct{}) error
+}
+
+type DefaultPortForwarder struct {
+	client *occlient.Client
+	k8sgenclioptions.IOStreams
+}
+
+func NewDefaultPortForwarder(client *occlient.Client, streams k8sgenclioptions.IOStreams) *DefaultPortForwarder {
+	return &DefaultPortForwarder{
+		client:    client,
+		IOStreams: streams,
+	}
+}
+
+func (f *DefaultPortForwarder) ForwardPorts(method string, url *url.URL, ports []string, stopChan, readyChan chan struct{}) error {
```

Comment:
Please add some commends most importantly explain what the arguments are.

Reply:
added

Reply:
shat are ports? In what form it expects the strings to be? <- this needs to be explained in the comments

Reply:
added

Reply:
thx